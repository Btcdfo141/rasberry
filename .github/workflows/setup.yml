name: Development environment setup

# yamllint disable-line rule:truthy
on:
  workflow_dispatch:
  schedule:
    - cron: "0 4 * * *"
  push:
    branches:
      - dev
      - rc
    paths:
      - "requirements.txt"
      - "requirements_all.txt"

jobs:
  setup:
    name: Initialize wheels builder
    strategy:
      fail-fast: false
      matrix:
        os: [ubuntu-latest, macos-latest]
    runs-on: ${{ matrix.os }}
    steps:
      - name: Checkout the repository
        uses: actions/checkout@v3.2.0

      - name: Run setup
        run: script/setup
