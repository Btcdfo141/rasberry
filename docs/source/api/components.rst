:mod:`spencerassistant.components`
===============================

air\_quality
--------------------------------------------

.. automodule:: spencerassistant.components.air_quality
    :members:
    :undoc-members:
    :show-inheritance:

alarm\_control\_panel
--------------------------------------------

.. automodule:: spencerassistant.components.alarm_control_panel
    :members:
    :undoc-members:
    :show-inheritance:

binary\_sensor
--------------------------------------------

.. automodule:: spencerassistant.components.binary_sensor
    :members:
    :undoc-members:
    :show-inheritance:

camera
---------------------------

.. automodule:: spencerassistant.components.camera
    :members:
    :undoc-members:
    :show-inheritance:

calendar
---------------------------

.. automodule:: spencerassistant.components.calendar
    :members:
    :undoc-members:
    :show-inheritance:

climate
---------------------------

.. automodule:: spencerassistant.components.climate
    :members:
    :undoc-members:
    :show-inheritance:

conversation
---------------------------

.. automodule:: spencerassistant.components.conversation
    :members:
    :undoc-members:
    :show-inheritance:

cover
---------------------------

.. automodule:: spencerassistant.components.cover
    :members:
    :undoc-members:
    :show-inheritance:

device\_tracker
---------------------------

.. automodule:: spencerassistant.components.device_tracker
    :members:
    :undoc-members:
    :show-inheritance:

fan
---------------------------

.. automodule:: spencerassistant.components.fan
    :members:
    :undoc-members:
    :show-inheritance:

light
---------------------------

.. automodule:: spencerassistant.components.light
    :members:
    :undoc-members:
    :show-inheritance:

lock
---------------------------

.. automodule:: spencerassistant.components.lock
    :members:
    :undoc-members:
    :show-inheritance:

media\_player
---------------------------

.. automodule:: spencerassistant.components.media_player
    :members:
    :undoc-members:
    :show-inheritance:

notify
---------------------------

.. automodule:: spencerassistant.components.notify
    :members:
    :undoc-members:
    :show-inheritance:

remote
---------------------------

.. automodule:: spencerassistant.components.remote
    :members:
    :undoc-members:
    :show-inheritance:

switch
---------------------------

.. automodule:: spencerassistant.components.switch
    :members:
    :undoc-members:
    :show-inheritance:

sensor
-------------------------------------

.. automodule:: spencerassistant.components.sensor
    :members:
    :undoc-members:
    :show-inheritance:

vacuum
-------------------------------------

.. automodule:: spencerassistant.components.vacuum
    :members:
    :undoc-members:
    :show-inheritance:

water\_heater
-------------------------------------

.. automodule:: spencerassistant.components.water_heater
    :members:
    :undoc-members:
    :show-inheritance:

weather
---------------------------

.. automodule:: spencerassistant.components.weather
    :members:
    :undoc-members:
    :show-inheritance:

webhook
---------------------------

.. automodule:: spencerassistant.components.webhook
    :members:
    :undoc-members:
    :show-inheritance:
