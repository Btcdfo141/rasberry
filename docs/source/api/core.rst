.. _core_module:

:mod:`spencerassistant.core`
-------------------------

.. automodule:: spencerassistant.core
    :members: