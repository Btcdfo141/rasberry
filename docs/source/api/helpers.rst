:mod:`spencerassistant.helpers`
============================

.. automodule:: spencerassistant.helpers
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.aiohttp\_client
-------------------------------------

.. automodule:: spencerassistant.helpers.aiohttp_client
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.area\_registry
------------------------------------

.. automodule:: spencerassistant.helpers.area_registry
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.check\_config
-----------------------------------

.. automodule:: spencerassistant.helpers.check_config
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.collection
--------------------------------

.. automodule:: spencerassistant.helpers.collection
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.condition
-------------------------------

.. automodule:: spencerassistant.helpers.condition
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.config\_entry\_flow
-----------------------------------------

.. automodule:: spencerassistant.helpers.config_entry_flow
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.config\_entry\_oauth2\_flow
-------------------------------------------------

.. automodule:: spencerassistant.helpers.config_entry_oauth2_flow
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.config\_validation
----------------------------------------

.. automodule:: spencerassistant.helpers.config_validation
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.data\_entry\_flow
---------------------------------------

.. automodule:: spencerassistant.helpers.data_entry_flow
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.debounce
------------------------------

.. automodule:: spencerassistant.helpers.debounce
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.deprecation
---------------------------------

.. automodule:: spencerassistant.helpers.deprecation
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.device\_registry
--------------------------------------

.. automodule:: spencerassistant.helpers.device_registry
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.discovery
-------------------------------

.. automodule:: spencerassistant.helpers.discovery
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.dispatcher
--------------------------------

.. automodule:: spencerassistant.helpers.dispatcher
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity
----------------------------

.. automodule:: spencerassistant.helpers.entity
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_component
---------------------------------------

.. automodule:: spencerassistant.helpers.entity_component
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_platform
--------------------------------------

.. automodule:: spencerassistant.helpers.entity_platform
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_registry
--------------------------------------

.. automodule:: spencerassistant.helpers.entity_registry
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_values
------------------------------------

.. automodule:: spencerassistant.helpers.entity_values
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entityfilter
----------------------------------

.. automodule:: spencerassistant.helpers.entityfilter
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.event
---------------------------

.. automodule:: spencerassistant.helpers.event
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.icon
--------------------------

.. automodule:: spencerassistant.helpers.icon
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.integration\_platform
-------------------------------------------

.. automodule:: spencerassistant.helpers.integration_platform
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.intent
----------------------------

.. automodule:: spencerassistant.helpers.intent
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.json
--------------------------

.. automodule:: spencerassistant.helpers.json
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.location
------------------------------

.. automodule:: spencerassistant.helpers.location
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.logging
-----------------------------

.. automodule:: spencerassistant.helpers.logging
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.network
-----------------------------

.. automodule:: spencerassistant.helpers.network
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.restore\_state
------------------------------------

.. automodule:: spencerassistant.helpers.restore_state
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.script
----------------------------

.. automodule:: spencerassistant.helpers.script
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.service
-----------------------------

.. automodule:: spencerassistant.helpers.service
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.signal
-----------------------------

.. automodule:: spencerassistant.helpers.signal
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.state
---------------------------

.. automodule:: spencerassistant.helpers.state
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.storage
-----------------------------

.. automodule:: spencerassistant.helpers.storage
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.sun
-------------------------

.. automodule:: spencerassistant.helpers.sun
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.system\_info
----------------------------------

.. automodule:: spencerassistant.helpers.system_info
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.temperature
---------------------------------

.. automodule:: spencerassistant.helpers.temperature
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.template
------------------------------

.. automodule:: spencerassistant.helpers.template
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.translation
---------------------------------

.. automodule:: spencerassistant.helpers.translation
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.typing
----------------------------

.. automodule:: spencerassistant.helpers.typing
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.update\_coordinator
-----------------------------------------

.. automodule:: spencerassistant.helpers.update_coordinator
   :members:
   :undoc-members:
   :show-inheritance:
