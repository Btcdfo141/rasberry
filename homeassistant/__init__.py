"""Init file for Home Assistant."""


test2
