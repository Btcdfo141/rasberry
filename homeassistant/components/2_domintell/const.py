"""Constants for the Domintell Integration integration."""

DOMAIN = "2_domintell"
