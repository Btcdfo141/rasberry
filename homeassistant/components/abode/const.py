"""Constants for the Abode Security System component."""
import logging

LOGGER = logging.getLogger(__package__)

DOMAIN = "abode"
ATTRIBUTION = "Data provided by goabode.com"

DEFAULT_CACHEDB = "jaraco_abode_cache.pickle"
CONF_POLLING = "polling"
