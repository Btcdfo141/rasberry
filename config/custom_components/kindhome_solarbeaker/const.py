from datetime import timedelta

from homeassistant.const import Platform

DOMAIN="kindhome_solarbeaker"
PLATFORMS = [Platform.COVER]

UPDATE_INTERVAL = timedelta(seconds=20)

SERVICE_UUID = "75c276c3-8f97-20bc-a143-b354244886d4"
TITLE="Kindhome Solarbeaker"

DATA_DEVICE = "device"
DATA_COOR = "coordinator"
