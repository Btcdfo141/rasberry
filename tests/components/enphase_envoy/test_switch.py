"""Test Enphase Envoy switch."""

from unittest.mock import AsyncMock, patch

import pytest
from syrupy.assertion import SnapshotAssertion

from homeassistant.components.enphase_envoy import DOMAIN
from homeassistant.components.enphase_envoy.const import Platform
from homeassistant.const import (
    ATTR_ENTITY_ID,
    SERVICE_TOGGLE,
    SERVICE_TURN_OFF,
    SERVICE_TURN_ON,
    STATE_OFF,
    STATE_ON,
)
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er
from homeassistant.setup import async_setup_component

from tests.common import ConfigType, MockConfigEntry


@pytest.fixture(name="setup_enphase_envoy_switch")
async def setup_enphase_envoy_switch_fixture(
    hass: HomeAssistant, config: ConfigType, mock_envoy: AsyncMock
):
    """Define a fixture to set up Enphase Envoy with number switch only."""
    with (
        patch(
            "homeassistant.components.enphase_envoy.config_flow.Envoy",
            return_value=mock_envoy,
        ),
        patch(
            "homeassistant.components.enphase_envoy.Envoy",
            return_value=mock_envoy,
        ),
        patch(
            "homeassistant.components.enphase_envoy.PLATFORMS",
            [Platform.SWITCH],
        ),
    ):
        assert await async_setup_component(hass, DOMAIN, config)
        await hass.async_block_till_done()
        yield


async def test_switch(
    hass: HomeAssistant,
    config_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
    setup_enphase_envoy_switch: AsyncMock,
    mock_envoy: AsyncMock,
) -> None:
    """Test enphase_envoy switch entities."""

    # these entities states should be created enabled from test data
    assert len(hass.states.async_all()) == 5

    entity_registry = er.async_get(hass)
    assert entity_registry

    # compare registered entities against snapshot of prior run
    entity_entries = er.async_entries_for_config_entry(
        entity_registry, config_entry.entry_id
    )
    assert entity_entries
    assert entity_entries == snapshot

    # Test if all entities still have same state
    for entity_entry in entity_entries:
        assert hass.states.get(entity_entry.entity_id) == snapshot(
            name=f"{entity_entry.entity_id}-state"
        )


@pytest.mark.parametrize(
    ("entity_id", "initial_value"),
    [
        ("switch.enpower_654321_grid_enabled", STATE_ON),
    ],
)
async def test_switch_grid_operation(
    hass: HomeAssistant,
    config_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
    setup_enphase_envoy_switch: AsyncMock,
    mock_envoy: AsyncMock,
    mock_go_on_grid: AsyncMock,
    mock_go_off_grid: AsyncMock,
    mock_close_dry_contact: AsyncMock,
    mock_open_dry_contact: AsyncMock,
    mock_disable_charge_from_grid: AsyncMock,
    mock_enable_charge_from_grid: AsyncMock,
    entity_id: str,
    initial_value: str,
) -> None:
    """Test enphase_envoy switch entities operation."""

    # verify initial value
    switch_state = hass.states.get(entity_id)
    assert switch_state.state == initial_value

    # build switching orders
    INITIAL_OFF_ORDER = (SERVICE_TURN_ON, SERVICE_TURN_OFF, SERVICE_TOGGLE)
    INITIAL_ON_ORDER = (SERVICE_TURN_OFF, SERVICE_TURN_ON, SERVICE_TOGGLE)

    # test each option for relays
    for option in INITIAL_OFF_ORDER if initial_value == STATE_OFF else INITIAL_ON_ORDER:
        await hass.services.async_call(
            Platform.SWITCH,
            option,
            {ATTR_ENTITY_ID: entity_id},
            blocking=True,
        )

    assert mock_go_on_grid.await_count == (2 if initial_value == STATE_OFF else 1)
    assert mock_go_off_grid.await_count == (2 if initial_value == STATE_ON else 1)

    # these should not been awaited
    mock_close_dry_contact.assert_not_called()
    mock_open_dry_contact.assert_not_called()
    mock_enable_charge_from_grid.assert_not_called()
    mock_disable_charge_from_grid.assert_not_called()


@pytest.mark.parametrize(
    ("entity_id", "initial_value"),
    [
        ("switch.enpower_654321_charge_from_grid", STATE_ON),
    ],
)
async def test_switch_grid_charge(
    hass: HomeAssistant,
    config_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
    setup_enphase_envoy_switch: AsyncMock,
    mock_envoy: AsyncMock,
    mock_go_on_grid: AsyncMock,
    mock_go_off_grid: AsyncMock,
    mock_close_dry_contact: AsyncMock,
    mock_open_dry_contact: AsyncMock,
    mock_disable_charge_from_grid: AsyncMock,
    mock_enable_charge_from_grid: AsyncMock,
    entity_id: str,
    initial_value: str,
) -> None:
    """Test enphase_envoy switch entities operation."""

    # verify initial value
    switch_state = hass.states.get(entity_id)
    assert switch_state.state == initial_value

    # build switching orders
    INITIAL_OFF_ORDER = (SERVICE_TURN_ON, SERVICE_TURN_OFF, SERVICE_TOGGLE)
    INITIAL_ON_ORDER = (SERVICE_TURN_OFF, SERVICE_TURN_ON, SERVICE_TOGGLE)

    # test each option for relays
    for option in INITIAL_OFF_ORDER if initial_value == STATE_OFF else INITIAL_ON_ORDER:
        await hass.services.async_call(
            Platform.SWITCH,
            option,
            {ATTR_ENTITY_ID: entity_id},
            blocking=True,
        )

    assert mock_enable_charge_from_grid.await_count == (
        2 if initial_value == STATE_OFF else 1
    )
    assert mock_disable_charge_from_grid.await_count == (
        2 if initial_value == STATE_ON else 1
    )

    # these should not been awaited
    mock_close_dry_contact.assert_not_called()
    mock_open_dry_contact.assert_not_called()
    mock_go_on_grid.assert_not_called()
    mock_go_off_grid.assert_not_called()


@pytest.mark.parametrize(
    ("entity_id", "initial_value"),
    [
        ("switch.envoy_1234", STATE_OFF),
        ("switch.envoy_1234_2", STATE_ON),
        ("switch.envoy_1234_3", STATE_OFF),
    ],
)
async def test_switch_relay_operation(
    hass: HomeAssistant,
    config_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
    setup_enphase_envoy_switch: AsyncMock,
    mock_envoy: AsyncMock,
    mock_go_on_grid: AsyncMock,
    mock_go_off_grid: AsyncMock,
    mock_close_dry_contact: AsyncMock,
    mock_open_dry_contact: AsyncMock,
    mock_disable_charge_from_grid: AsyncMock,
    mock_enable_charge_from_grid: AsyncMock,
    entity_id: str,
    initial_value: str,
) -> None:
    """Test enphase_envoy switch entities operation."""

    # verify initial value
    switch_state = hass.states.get(entity_id)
    assert switch_state.state == initial_value

    # build switching orders
    INITIAL_OFF_ORDER = (SERVICE_TURN_ON, SERVICE_TURN_OFF, SERVICE_TOGGLE)
    INITIAL_ON_ORDER = (SERVICE_TURN_OFF, SERVICE_TURN_ON, SERVICE_TOGGLE)

    # test each option for relays
    for option in INITIAL_OFF_ORDER if initial_value == STATE_OFF else INITIAL_ON_ORDER:
        await hass.services.async_call(
            Platform.SWITCH,
            option,
            {ATTR_ENTITY_ID: entity_id},
            blocking=True,
        )

    assert mock_close_dry_contact.await_count == (
        2 if initial_value == STATE_OFF else 1
    )
    assert mock_open_dry_contact.await_count == (2 if initial_value == STATE_ON else 1)

    # these should not been awaited
    mock_go_off_grid.assert_not_called()
    mock_go_on_grid.assert_not_called()
    mock_enable_charge_from_grid.assert_not_called()
    mock_disable_charge_from_grid.assert_not_called()
