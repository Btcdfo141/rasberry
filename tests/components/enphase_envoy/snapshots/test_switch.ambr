# serializer version: 1
# name: test_switch
  list([
    EntityRegistryEntrySnapshot({
      'aliases': set({
      }),
      'area_id': None,
      'capabilities': None,
      'config_entry_id': <ANY>,
      'device_class': None,
      'device_id': <ANY>,
      'disabled_by': None,
      'domain': 'switch',
      'entity_category': None,
      'entity_id': 'switch.enpower_654321_grid_enabled',
      'has_entity_name': True,
      'hidden_by': None,
      'icon': None,
      'id': <ANY>,
      'labels': set({
      }),
      'name': None,
      'options': dict({
      }),
      'original_device_class': None,
      'original_icon': None,
      'original_name': 'Grid enabled',
      'platform': 'enphase_envoy',
      'previous_unique_id': None,
      'supported_features': 0,
      'translation_key': 'grid_enabled',
      'unique_id': '654321_mains_admin_state',
      'unit_of_measurement': None,
    }),
    EntityRegistryEntrySnapshot({
      'aliases': set({
      }),
      'area_id': None,
      'capabilities': None,
      'config_entry_id': <ANY>,
      'device_class': None,
      'device_id': <ANY>,
      'disabled_by': None,
      'domain': 'switch',
      'entity_category': None,
      'entity_id': 'switch.envoy_1234',
      'has_entity_name': True,
      'hidden_by': None,
      'icon': None,
      'id': <ANY>,
      'labels': set({
      }),
      'name': None,
      'options': dict({
      }),
      'original_device_class': None,
      'original_icon': None,
      'original_name': None,
      'platform': 'enphase_envoy',
      'previous_unique_id': None,
      'supported_features': 0,
      'translation_key': None,
      'unique_id': '654321_relay_NC1_relay_status',
      'unit_of_measurement': None,
    }),
    EntityRegistryEntrySnapshot({
      'aliases': set({
      }),
      'area_id': None,
      'capabilities': None,
      'config_entry_id': <ANY>,
      'device_class': None,
      'device_id': <ANY>,
      'disabled_by': None,
      'domain': 'switch',
      'entity_category': None,
      'entity_id': 'switch.envoy_1234_2',
      'has_entity_name': True,
      'hidden_by': None,
      'icon': None,
      'id': <ANY>,
      'labels': set({
      }),
      'name': None,
      'options': dict({
      }),
      'original_device_class': None,
      'original_icon': None,
      'original_name': None,
      'platform': 'enphase_envoy',
      'previous_unique_id': None,
      'supported_features': 0,
      'translation_key': None,
      'unique_id': '654321_relay_NC2_relay_status',
      'unit_of_measurement': None,
    }),
    EntityRegistryEntrySnapshot({
      'aliases': set({
      }),
      'area_id': None,
      'capabilities': None,
      'config_entry_id': <ANY>,
      'device_class': None,
      'device_id': <ANY>,
      'disabled_by': None,
      'domain': 'switch',
      'entity_category': None,
      'entity_id': 'switch.envoy_1234_3',
      'has_entity_name': True,
      'hidden_by': None,
      'icon': None,
      'id': <ANY>,
      'labels': set({
      }),
      'name': None,
      'options': dict({
      }),
      'original_device_class': None,
      'original_icon': None,
      'original_name': None,
      'platform': 'enphase_envoy',
      'previous_unique_id': None,
      'supported_features': 0,
      'translation_key': None,
      'unique_id': '654321_relay_NC3_relay_status',
      'unit_of_measurement': None,
    }),
    EntityRegistryEntrySnapshot({
      'aliases': set({
      }),
      'area_id': None,
      'capabilities': None,
      'config_entry_id': <ANY>,
      'device_class': None,
      'device_id': <ANY>,
      'disabled_by': None,
      'domain': 'switch',
      'entity_category': None,
      'entity_id': 'switch.enpower_654321_charge_from_grid',
      'has_entity_name': True,
      'hidden_by': None,
      'icon': None,
      'id': <ANY>,
      'labels': set({
      }),
      'name': None,
      'options': dict({
      }),
      'original_device_class': None,
      'original_icon': None,
      'original_name': 'Charge from grid',
      'platform': 'enphase_envoy',
      'previous_unique_id': None,
      'supported_features': 0,
      'translation_key': 'charge_from_grid',
      'unique_id': '654321_charge_from_grid',
      'unit_of_measurement': None,
    }),
  ])
# ---
# name: test_switch[switch.enpower_654321_charge_from_grid-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Enpower 654321 Charge from grid',
    }),
    'context': <ANY>,
    'entity_id': 'switch.enpower_654321_charge_from_grid',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_switch[switch.enpower_654321_grid_enabled-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Enpower 654321 Grid enabled',
    }),
    'context': <ANY>,
    'entity_id': 'switch.enpower_654321_grid_enabled',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_switch[switch.envoy_1234-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Envoy 1234',
    }),
    'context': <ANY>,
    'entity_id': 'switch.envoy_1234',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_switch[switch.envoy_1234_2-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Envoy 1234',
    }),
    'context': <ANY>,
    'entity_id': 'switch.envoy_1234_2',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_switch[switch.envoy_1234_3-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Envoy 1234',
    }),
    'context': <ANY>,
    'entity_id': 'switch.envoy_1234_3',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
